enterprise/server/server_notification/server_notification.go

```diff
@@ -0,0 +1,127 @@
+// server_notification provides a best-effort pubsub-style notification service
+// for broadcasting information to other servers within the same service.
+//
+// Payloads are defined by adding a new proto to the Notification parent
+// proto in the server_notification proto file.
+package server_notification
+
+import (
+	"context"
+	"encoding/base64"
+	"sync"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/backends/pubsub"
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/util/alert"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/go-redis/redis/v8"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+
+	snpb "github.com/buildbuddy-io/buildbuddy/proto/server_notification"
+)
+
+const (
+	baseChannelName = "serverNotifications/"
+)
+
+type Service struct {
+	ps          *pubsub.PubSub
+	channelName string
+
+	msgTypes map[protoreflect.MessageDescriptor]protoreflect.FieldDescriptor
+
+	mu   sync.Mutex
+	subs map[protoreflect.MessageDescriptor][]chan<- proto.Message
+}
+
+func Register(env environment.Env, serviceName string) error {
+	rdb := env.GetDefaultRedisClient()
+	if rdb == nil {
+		return nil
+	}
+
+	env.SetServerNotificationService(New(serviceName, rdb))
+	return nil
+}
+
+func New(serviceName string, rdb redis.UniversalClient) *Service {
+	ps := pubsub.NewPubSub(rdb)
+
+	msgTypes := make(map[protoreflect.MessageDescriptor]protoreflect.FieldDescriptor)
```

Comment:
In practice everything is going to use proto so I don't see any benefit of switching to bytes. If we switch away from using the Notification parent message, we'd need to introduce another way to indicate the type of payload so that the receiver can dispatch the messages properly. Using the proto message type as the discriminator seems pretty simple and intuitive. I also don't expect this mechanism to be used in a significant number of places so I don't think bazel deps is a big concern here.